src/AudioFileSourceHTTPStream.h

```diff
@@ -52,14 +56,42 @@ class AudioFileSourceHTTPStream : public AudioFileSource
     enum { STATUS_HTTPFAIL=2, STATUS_DISCONNECTED, STATUS_RECONNECTING, STATUS_RECONNECTED, STATUS_NODATA };
 
   private:
-    virtual uint32_t readInternal(void *data, uint32_t len, bool nonBlock);
+    bool is_chunked;
+    std::size_t next_chunk;
     WiFiClient client;
     HTTPClient http;
     int pos;
     int size;
     int reconnectTries;
     int reconnectDelayMs;
     char saveURL[128];
+    uint32_t (AudioFileSourceHTTPStream::*readImpl)(void *data, uint32_t len, bool nonBlock);
```

Comment:
I think the `AudioFileSourceHTTPStream::` is superfluous here.  We're already in `class AudioFileSourceHTTPStream`...

Reply:
From what I've checked, this is how it can be used. You're welcomed to further check it yourself.